File: Ex-16/ex_16_java/Cliente.java
public class Cliente {
    private String nome;
    private String cpf;
    // Atributo do link para a associação unidirecional com Conta
    private Conta[] contas;
    private int indiceContas; 

    /**
     * Construtora para iniciar os atributos do cliente.
     * @param nome O nome do cliente.
     * @param cpf O CPF do cliente.
     */
    public Cliente(String nome, String cpf) {
        this.nome = nome;
        this.cpf = cpf;
        this.contas = new Conta[3]; // Cliente pode ter até 3 contas
        this.indiceContas = 0;
    }

    /**
     * Adiciona uma conta ao cliente, se houver espaço.
     * @param conta A conta a ser adicionada.
     */
    public void adicionarConta(Conta conta) {
        if (indiceContas < 3) {
            this.contas[indiceContas] = conta;
            this.indiceContas++;
            System.out.println("Conta " + conta.getNumero() + " adicionada ao cliente " + this.nome);
        } else {
            System.out.println("Limite de contas atingido para o cliente " + this.nome);
        }
    }

    /**
     * Aplica um valor de recursos (depósito) em todas as contas do cliente.
     * @param valor O valor a ser depositado em cada conta.
     */
    public void aplicarRecursos(int valor) {
        System.out.println("\n-- Aplicando recursos de R$" + valor + " para o cliente " + this.nome + " --");
        for (int i = 0; i < indiceContas; i++) {
            contas[i].depositar(valor);
        }
    }

    /**
     * Gera um relatório com os dados do cliente e o saldo de suas contas.
     * @return Uma String com o relatório completo.
     */
    public String gerarRelatorio() {
        StringBuilder relatorio = new StringBuilder();
        relatorio.append("========================================\n");
        relatorio.append("Relatório do Cliente\n");
        relatorio.append("========================================\n");
        relatorio.append("Nome: ").append(this.nome).append("\n");
        relatorio.append("CPF: ").append(this.cpf).append("\n\n");
        relatorio.append("--- Contas Associadas ---\n");

        if (indiceContas == 0) {
            relatorio.append("Nenhuma conta associada.\n");
        } else {
            for (int i = 0; i < indiceContas; i++) {
                relatorio.append("Conta Número: ").append(contas[i].getNumero()).append("\n");
                relatorio.append(String.format("Saldo: R$%.2f\n", contas[i].getSaldo()));
                relatorio.append("----------------------------------------\n");
            }
        }
        return relatorio.toString();
    }
}

File: Ex-16/ex_16_python/Cliente.py
class Cliente:
    # Constante com o nro máximo de contas
    N_MAX_CONTAS = 3

    # Construtora & Destrutora
    def __init__(self, nome, cpf):
        self.nome = nome
        self.cpf = cpf
        self.nContas = 0
        # Cria vetor para as contas
        self.contas = [None] * Cliente.N_MAX_CONTAS
        # Inicializamos todas as posicoes
        for i in range(Cliente.N_MAX_CONTAS):
            self.contas[i] = None

    def adicionarConta(self, conta):
        if self.nContas < Cliente.N_MAX_CONTAS:
            # Adiciona nova conta
            self.contas[self.nContas] = conta
            # Aumenta nro de contas
            self.nContas += 1

    # Métodos requeridos
    def gerarRelatorio(self):
        # Cria o string
        ss = self.nome + " | " + self.cpf + " | N. Contas: " + str(self.nContas)

        for i in range(self.nContas):
            ss += " | Saldo Conta[" + str(i) + "]<" + str(self.contas[i].getNumero()) + ">: " + str(self.contas[i].getSaldo())

        return ss

    def aplicarRecursos(self, valor, iConta=0):
        # Verifica que a posicao da conta seja valida
        if self.nContas > iConta:
            self.contas[iConta].depositar(valor)
